Use of “int32_t” type causes build failure on ARMv6

In `dr_mp3.h`, the `int32_t` type is used on ARMv6:

```c
#if defined(__ARM_ARCH) && (__ARM_ARCH >= 6) && !defined(__aarch64__) && !defined(_M_ARM64)
#define DRMP3_HAVE_ARMV6 1
static __inline__ __attribute__((always_inline)) drmp3_int32 drmp3_clip_int16_arm(int32_t a)
{
    drmp3_int32 x = 0;
    __asm__ ("ssat %0, #16, %1" : "=r"(x) : "r"(a));
    return x;
}
#else
#define DRMP3_HAVE_ARMV6 0
#endif
```

Since the library is C89 without `stdint.h`/`inttypes.h`, this type is unknown and the build fails:

```
In file included from ./tests/mp3/dr_mp3_test_0.c:2:
./tests/mp3/../../dr_mp3.h:704:83: error: unknown type name 'int32_t'
  704 | static __inline__ __attribute__((always_inline)) drmp3_int32 drmp3_clip_int16_arm(int32_t a)
      |                                                                                   ^~~~~~~
In file included from ./tests/mp3/dr_mp3_test_0.c:2:
./tests/mp3/../../dr_mp3.h: In function 'drmp3d_scale_pcm':
./tests/mp3/../../dr_mp3.h:1937:22: warning: implicit declaration of function 'drmp3_clip_int16_arm' [-Wimplicit-function-declaration]
 1937 |     s = (drmp3_int16)drmp3_clip_int16_arm(s32);
      |                      ^~~~~~~~~~~~~~~~~~~~
```

The appropriate fix seems to be to replace the `int32_t` parameter with an `drmp3_int32` one; this type is already used in the function body and the function return value. I have confirmed this solves the problem, so I will follow up with a PR.

--- a/dr_mp3.h
+++ b/dr_mp3.h
@@ -701,7 +701,7 @@ static int drmp3_have_simd(void)
 
 #if defined(__ARM_ARCH) && (__ARM_ARCH >= 6) && !defined(__aarch64__) && !defined(_M_ARM64)
 #define DRMP3_HAVE_ARMV6 1
-static __inline__ __attribute__((always_inline)) drmp3_int32 drmp3_clip_int16_arm(int32_t a)
+static __inline__ __attribute__((always_inline)) drmp3_int32 drmp3_clip_int16_arm(drmp3_int32 a)
 {
     drmp3_int32 x = 0;
     __asm__ ("ssat %0, #16, %1" : "=r"(x) : "r"(a));

